ethsigner/signer/azure/src/main/java/tech/pegasys/ethsigner/signer/azure/AzureKeyVaultTransactionSignerFactory.java

```diff
@@ -40,23 +40,23 @@
   private static final Logger LOG = LogManager.getLogger();
 
   private static final String AZURE_URL_PATTERN = "https://%s.vault.azure.net";
-
-  private final KeyVaultClientCustom client;
-  private final String baseUrl;
+  private final AzureKeyVaultAuthenticator vaultAuthenticator;
 
   public AzureKeyVaultTransactionSignerFactory(
-      final String keyVaultName, final KeyVaultClientCustom client) {
-    this.client = client;
-    this.baseUrl = constructAzureKeyVaultUrl(keyVaultName);
+      final AzureKeyVaultAuthenticator vaultAuthenticator) {
+    this.vaultAuthenticator = vaultAuthenticator;
   }
 
-  public TransactionSigner createSigner(final String keyName, final String keyVersion) {
-    checkNotNull(keyName, "keyName must be specified");
+  public TransactionSigner createSigner(final AzureConfig config) {
+    checkNotNull(config, "Config must be specified");
+    final KeyVaultClientCustom client =
+        vaultAuthenticator.getAuthenticatedClient(config.getClientId(), config.getClientSecret());
+    final String baseUrl = constructAzureKeyVaultUrl(config.getKeyvaultName());
 
     final JsonWebKey key;
     final KeyIdentifier kid;
     try {
-      kid = new KeyIdentifier(baseUrl, keyName, keyVersion);
+      kid = new KeyIdentifier(baseUrl, config.getKeyName(), config.getKeyVersion());
```

Comment:
nit: kId?

Reply:
updated to keyIdentifier (carry over from Kaleido code